app/src/main/java/com/example/myapplication/adapter/UsersAdapter.kt

```diff
@@ -44,15 +88,14 @@ class UsersAdapter(private val actionListener: UserActionListener) :
         }
     }
 
-
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UsersViewHolder {
         val inflater = LayoutInflater.from(parent.context)
         val binding = UserPatternBinding.inflate(inflater, parent, false)
-        return UsersViewHolder(binding, actionListener)
+        return UsersViewHolder(binding, actionListener, isModeActive, this::changeModeStatus)
```

Comment:
You can declare viewholder class as inner class and get access to adapter class properties in viewholder class. Hence you will not need pass lambda as constructor parameter.